ci/cloudbuild/run_test.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+#
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# `-e` enables the script to automatically fail when a command fails
+# `-o pipefail` sets the exit code to the rightmost comment to exit
+# with a non-zero
+set -eo pipefail
+
+# In cloudbuild, the current directory is the project root.
+export PROJECT_ROOT=$(pwd)
+
+# A script file for running the test in a sub project.
+test_script="${PROJECT_ROOT}/ci/cloudbuild/run_single_test.sh"
+
+if [ ${BUILD_TYPE} == "presubmit" ]; then
+
+    # For presubmit build, we want to know the difference from the
+    # common commit in origin/main.
+    GIT_DIFF_ARG="origin/main..."
```

Comment:
Is it possible to make collecting this branch name a bit smarter, in case we're working off a different branch?

Reply:
It is technically possible with the public API I think. IMO it's not worth our time because we don't have a plan to change the default branch name.